src/webapp/components/dataset-table/DataSetTableConfig.tsx

```diff
@@ -136,20 +136,27 @@ export type CommonActionsProps<T> = {
     user: User;
 };
 
-export function getCommonActions<T extends ReferenceObject>(
+export interface HasPermission extends ReferenceObject {
+    hasPermissionsToUpdate(user: User): boolean;
+}
+
+export function getCommonActions<T extends HasPermission>(
     props: CommonActionsProps<T>
 ): DataTableAction<T>[] {
     const { onAction, navigateTo, user } = props;
+
+    const isActionActive = (records: T[]): boolean => {
+        return props.isActive(records) && canDataSetBeUpdated(records, user);
+    };
```

Comment:
No need to change, just as a note. Sometimes is more declarative to name variables not by how they will be used, but what they represent by themselves. For example: `canDataSetsBeUpdated` or whatever.